pkg/trace/agent/agent.go

```diff
@@ -215,6 +215,12 @@ func (a *Agent) setRootSpanTags(root *pb.Span) {
 		rate := ratelimiter.RealRate()
 		sampler.SetPreSampleRate(root, rate)
 	}
+
+	if a.conf.InAzureAppServices {
```

Comment:
I don't think setting these only on root spans is correct for all cases. For example, when tracer are using "partial flushing" they can split a single trace into multiple payloads. Only one of those will contain the root span, but each payload must contain some of these tags. Let me check the .NET tracer code...

Reply:
Got it. The .NET tracer sets these tags on the root span (if available) or on any orphan spans in a trace payload (meaning a span whose parent is not present in the current payload because the trace was probably split). [See here.](https://github.com/DataDog/dd-trace-dotnet/blob/b74324fb3dbbf84afe903e7cf8d1ba695b9d61e1/tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs#L304)

In the agent, `setRootSpanTags` is called for the root span if it is found, or on one of the orphan spans, so doing it here is probably fine. Carry on :)

Reply:
Oh, wait. I knew there was a gotcha [somewhere](https://github.com/DataDog/dd-trace-dotnet/blob/b74324fb3dbbf84afe903e7cf8d1ba695b9d61e1/tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs#L381C17-L381C120):
```
// the front-end identify AAS spans using aas.site.name and aas.site.type, so we need them on all spans
```

Reply:
@lucaspimentel  this is a ticket we have had open for a while for Linux . We are only doing root spans which doesn't work well in the UI. Maybe we can tackle it now.

Reply:
> We are only doing root spans which doesn't work well in the UI

Right. We should confirm this with someone from intake/frontend, but my understanding is that the minimum requirements for AAS are:
- `aas.site.name` and `aas.site.type` must appear on all AAS spans for the AAS UI to be triggered correctly
- all the other AAS tags must appear in at least one span on every trace chunk. it can be any span or even multiple spans.
- all AAS tags should be on the root span as well so they can be used for indexing/search, and so they are easy to find for users in the UI

Adding the tags to more spans than is required won't break anything, but it makes the trace payloads larger, which affects performance and billing. Adding all the AAS to every spans is easy and covers all the bases, but should be avoided if possible. (it could be fine short term if we plan to come back and optimize later)

cc @pierotibou who has worked on this topic several times in the .NET tracer

Reply:
Just made the change to add all AAS tags to the root span but only the two required ones for all other spans